lib/fs.js

```diff
@@ -2039,28 +2044,24 @@ function lutimesSync(path, atime, mtime) {
   handleErrorFromBinding(ctx);
 }
 
-function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
+function writeAll(
+  fd, isUserFd, buffer, offset, length, signal, encoding, callback) {
   if (signal?.aborted) {
-    const abortError = new AbortError();
-    if (isUserFd) {
-      callback(abortError);
-    } else {
-      fs.close(fd, (err) => {
-        callback(aggregateTwoErrors(err, abortError));
+    handleWriteAllErrorCallback(fd, isUserFd, new AbortError(), callback);
+    return;
+  }
+
+  if (isCustomIterable(buffer)) {
+    writeAllCustomIterable(
+      fd, isUserFd, buffer, offset, length, signal, encoding, callback)
+      .catch((reason) => {
+        handleWriteAllErrorCallback(fd, isUserFd, reason, callback);
```

Comment:
I would schedule this via `process.nextTick`. If this throws synchronously for whatever reason, it will lead to an unhandled rejection.